core/domain/platform_parameter_domain.py

```diff
@@ -879,6 +864,19 @@ def evaluate(
             *. The evaluate result of the platform parameter.
         """
         if context.is_valid:
+            if self._is_feature:
+                server_mode = self._get_server_mode().value
+                if (
+                    server_mode == ServerMode.TEST.value and
+                    self._feature_stage == ServerMode.DEV.value
+                ):
+                    return False
+                if (
+                    server_mode == ServerMode.PROD.value and
+                    self._feature_stage in (
+                        ServerMode.DEV.value, ServerMode.TEST.value)
+                ):
+                    return False
```

Comment:
```suggestion
                server_mode = self._get_server_mode()
                if (
                    server_mode == ServerMode.TEST and
                    self._feature_stage == ServerMode.DEV
                ):
                    return False
                if (
                    server_mode == ServerMode.PROD and
                    self._feature_stage in {
                        ServerMode.DEV, ServerMode.TEST
                    }
                ):
                    return False
```

Shouldn't this work?

Reply:
ditto, for the same occurrence below

Reply:
It will not work because the `self._feature_stage` has type of str and we would require to get the value of enum's value. I have made the changes for the `server_mode` part.